From cc311b61b82b7510a3a6cb0952d3a331e3551e05 Mon Sep 17 00:00:00 2001
From: Eric Koegel <eric.koegel@gmail.com>
Date: Mon, 13 Jun 2016 15:08:11 +0300
Subject: [PATCH] Settings: Allow directories to be selected

Gtk3 isn't as bright as Gtk2 so we need to add directories to the
filter for the GtkFileChooser, otherwise you can't select
directories while in directory mode...
---
 settings/main.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/settings/main.c b/settings/main.c
index ce08110..39eeafa 100644
--- a/settings/main.c
+++ b/settings/main.c
@@ -1852,6 +1852,9 @@ xfdesktop_settings_dialog_setup_tabs(GtkBuilder *main_gxml,
     filter = gtk_file_filter_new();
     gtk_file_filter_set_name(filter, _("Image files"));
     gtk_file_filter_add_pixbuf_formats(filter);
+    gtk_file_filter_add_mime_type(filter, "inode/directory");
+    gtk_file_filter_add_mime_type(filter, "application/x-directory");
+    gtk_file_filter_add_mime_type(filter, "text/directory");
     gtk_file_chooser_add_filter(GTK_FILE_CHOOSER(panel->btn_folder), filter);
 
     /* Get default wallpaper folder */
-- 
2.9.3

